packages/amazon-sumerian-hosts-core/src/core/awspack/LexFeature.js

```diff
@@ -0,0 +1,287 @@
+ *
+ * @property {Object} LEX_DEFAULTS - Default values to use with calls to {@link external:LexRuntime}.
+ * @property {string} [LEX_DEFAULTS.SampleRate='16000']
+ * @property {Object} EVENTS - Built-in messages that the feature emits.
+ * @property {string} [EVENTS.lexResponse=onLexResponseEvent] - Message that is emitted after
+ * receiving lex response for the input sent
+ * @property {string} [EVENTS.micReady=onMicrophoneReadyEvent] - Message that is emitted after
+ * microphone is ready to use
+ * @property {string} [EVENTS.micReady=onMicrophoneBeginRecordingEvent] - Message that is emitted after
+ * microphone starts recording
+ * @property {string} [EVENTS.micReady=onMicrophoneEndRecordingEvent] - Message that is emitted after
+ * microphone ends recording
+ * @property {Object} SERVICES - AWS services that are necessary for the feature
+ * to function.
+ * @property {external:LexRuntime} SERVICES.lexRuntime - The LexRuntime service that is used
+ * to interact with lex bot. Will be undefined until [initializeService]{@link LexFeature.initializeService}
+ * has been successfully executed
+ */
+class LexFeature{
+  /**
+   * @constructor
+   *
+   * @param {Messenger=} messenger - Optional Messenger object which the event will be emitted through
+   * @param {Object=} options - Options that will be used to interact with lex bot.
+   * @param {string=} options.botName - The name of the lex bot.
+   * @param {string=} options.botAlias - The alias of the lex bot.
+   * @param {string=} options.userId - The userId used to keep track of the session with lex bot. 
+  */
+  constructor(
+    messenger = undefined,
+    options = {
+      botName: undefined,
+      botAlias: undefined,
+      userId: undefined,
+    }
+  ) {
+    this._messenger = messenger;
+
+    this._botName = options.botName;
+    this._botAlias = options.botAlias;
+    this._userId = options.userId ? options.userId : Utils.createId();
+
+    //Microphone related fields
+    this._recording = false;
+    this._recLength = 0;
+    this._recBuffer = [];
+    this._setAudioContext();
+  }
+
+  /**
+   * Store LexRuntime and AWS SDK Version for use across all instances.
+   *
+   * @param {external:LexRuntime} lexRuntime - lexRuntime instance to use to interact with lex bot.
+   */
+  static initializeService(lexRuntime) {
+    // Make sure all were defined
+    if (lexRuntime === undefined)
+    {
+      throw new Error(
+        'Cannot initialize Lex feature. LexRuntime must be defined'
+      );
+    }
+
+    // Add sumerian hosts user-agent
+    if (lexRuntime.config) {
+      lexRuntime.config.customUserAgent = Utils.withCustomUserAgent(
+        lexRuntime.config.customUserAgent
+      );
+    }
+
+    // Store parameters
+    this.SERVICES.lexRuntime = lexRuntime;
+  }
+
+  _setAudioContext() {
+    AudioContext = window.AudioContext || window.webkitAudioContext;
```

Comment:
Do we state how far back we support? what browsers does window.webkitAudioContext cover?

Reply:
Actually, i will just stick with AudioContext as TextToSpeechFeature only assumes modern browser i think